transcom/mymove
# [MB-16449] Fix the Total Segment Count displayed in the SE segment

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16449)

## Summary

This PR resolves an issue with the total segment count being off. I found that the invoice generator was using a hardcoded value of 7 to represent the `ServiceItemSegments` however, because the `FA2` field was previously updated to be an array, the 7 is no longer was always accurate. 

I was curious why the invoice service test didn't fail previously and I believe it's because the test code generated a single FA2 segment, so the hardcoded value of 7 was still accurate in that case. So for this PR, I created a test that generated multiple FA2 segments to check that both segments would be included in the total. 

This PR also updates how the total segment count is calculated to take into account the possibility of multiple FA2 segments


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Please refer to the [Acceptance Testing doc](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing) for EDI generation or more information re: `prime-api-client` commands

1. Go to the office app and Login as a TIO with gbloc KKFA
2. Search for move code "RDY4PY" and click it
3. Scroll down to payment requests, click Review, and approve 1+.
4. Refer to the [Acceptance Testing doc](https://transcom.github.io/mymove-docs/docs/backend/testing/acceptance-testing-syncada-edi-invoicing). 
5. In your terminal, if you have the json file from the doc in root dir, run `prime-api-client --insecure create-payment-request --filename ./payload.json | jq '.id,.paymentRequestNumber'` which should give you the 9 digit payment request ID for the following command
6. Run `make bin/prime-api-client bin/generate-payment-request-edi && generate-payment-request-edi --payment-request-number 6682-5285-1` replacing the end numbers with the payment request ID from previous
7. Confirm that the EDI prints in your terminal and does NOT include the the second FA2 field, as hhg SAC's empty. Note the SE count
8. Go back into the move and add HHG and NTS SACs
9. Run step 6 again and check that HHG SAC's in a new FA2 field and that the SE count takes the multiple FA2 fields into account


## Screenshots
#### Count without SAC/multiple FA2 segments 
<img width="430" alt="Screenshot 2023-07-03 at 10 29 04 AM" src="https://github.com/transcom/mymove/assets/67110378/44b0d837-96d4-4573-ab18-a3654e7e4be7">

#### Count with SAC/multiple FA2 segments 
<img width="504" alt="Screenshot 2023-07-03 at 10 30 21 AM" src="https://github.com/transcom/mymove/assets/67110378/2c852cee-b79e-4108-8dca-9f365dd2dc90">
